framework/mutex/Mutex.php

```diff
@@ -97,6 +103,33 @@ public function release($name)
         return false;
     }
 
+    /**
+     * Executes callback with mutex synchronization.
+     *
+     * @param string $name of the lock to be acquired. Must be unique.
+     * @param int $timeout time (in seconds) to wait for lock to be released.
+     * @param callable $callback a valid PHP callback that performs the job. Accepts mutex instance as parameter.
+     * @param bool $throw whether to throw an exception when the lock is not acquired.
+     * @return mixed result of callback function, or null when the lock is not acquired.
+     * @throws SyncException when the lock is not acquired.
+     * @since 2.1
```

Comment:
If so, you must create PR for 2.1 branch, **not for master**.

Reply:
@rugabarbo Mutex class is not changed in 2.1. The PR can be merged into 2.1 without conflicts.